src/sbvr-api/hooks.ts

```diff
@@ -0,0 +1,77 @@
+import * as _ from 'lodash'
+import * as Promise from 'bluebird'
+import { settleMapSeries } from './control-flow'
+
+type RollbackAction = () => any
+type HookFn = () => any
+type HookBluePrint = {
+	HOOK: HookFn
+	effects: boolean
+}
+export class Hook {
+	private hookFn: Function
+
+	constructor(hookFn: HookFn) {
```

Comment:
Continuing from https://github.com/resin-io/pinejs/pull/155#discussion_r181603674 I think the issue is you tried to make the change to `SideEffectHook`, it should be fine to add to just `Hook` (and is the only place it's needed anyway as it replaces the `this.x = ...`)